cluster/src/main/java/io/scalecube/cluster/membership/JmxMembershipProvider.java

```diff
@@ -0,0 +1,55 @@
+package io.scalecube.cluster.membership;
+
+
+import io.scalecube.cluster.Cluster;
+import io.scalecube.cluster.Member;
+
+import java.util.Collection;
+import java.util.List;
+import java.util.function.Predicate;
+import java.util.stream.Collectors;
+
+public class JmxMembershipProvider {
+  private final MembershipProtocolImpl membership;
+  private final Cluster cluster;
+
+  public JmxMembershipProvider(Cluster cluster,
+                               MembershipProtocolImpl membership) {
+    this.cluster = cluster;
+    this.membership = membership;
+
+  }
+
+
+  public int incarnation() {
+    Member current = cluster.member();
+    return membership.getMembershipRecords()
+                        .stream()
+                        .filter( rec -> rec.id().equals(current.id()) )
+                        .map( MembershipRecord::incarnation )
+                        .findFirst()
+                        .orElse(-1);
+  }
+
+  public List<String> aliveMembers() {
+    return findRecordsByCondition(MembershipRecord::isAlive);
+  }
+
+  public List<String> deadMembers() {
+    return findRecordsByCondition( MembershipRecord::isDead );
```

Comment:
When member is getting to _dead_ state, we remove it from membership table; you will not be able track later dead members by simply scanning membership table. I had in mind to create a new map inside membership protocol for dead members only, so monitoring tools could pick it up. Let's limit this map to 30 records, i.e. keep 30 recent dead members.